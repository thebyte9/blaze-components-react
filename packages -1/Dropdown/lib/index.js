"use strict";
var __importStar = (this && this.__importStar) || function (mod) {
    if (mod && mod.__esModule) return mod;
    var result = {};
    if (mod != null) for (var k in mod) if (Object.hasOwnProperty.call(mod, k)) result[k] = mod[k];
    result["default"] = mod;
    return result;
};
exports.__esModule = true;
var react_1 = __importStar(require("react"));
var Dropdown = function (_a) {
    var label = _a.label, children = _a.children;
    var _b = react_1.useState("close"), toggled = _b[0], setToggled = _b[1];
    var toggleMenu = function () {
        var menuStatus = toggled === "close" ? "open" : "close";
        setToggled(menuStatus);
    };
    return (react_1["default"].createElement(react_1.Fragment, null,
        react_1["default"].createElement("div", { className: "more-menu__wrapper" },
            react_1["default"].createElement("button", { onClick: toggleMenu, type: "button", className: "icon-button toggle" },
                label,
                react_1["default"].createElement("i", { className: "material-icons" }, "more_vert")),
            react_1["default"].createElement("div", { className: "more-menu " + toggled },
                react_1["default"].createElement("ul", { className: "more-menu__list" },
                    react_1["default"].createElement("li", { className: "more-menu__list-item" }, react_1["default"].Children.map(children, function (child) { return react_1["default"].cloneElement(child); })))))));
};
Dropdown.defaultProps = {
    children: [],
    label: "Menu"
};
exports["default"] = Dropdown;
