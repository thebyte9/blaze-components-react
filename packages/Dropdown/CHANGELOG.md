# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

# [0.8.0-alpha.22](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.21...v0.8.0-alpha.22) (2021-07-19)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.21](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.20...v0.8.0-alpha.21) (2021-07-14)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.20](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.19...v0.8.0-alpha.20) (2021-07-13)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.19](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.18...v0.8.0-alpha.19) (2021-07-13)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.18](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.17...v0.8.0-alpha.18) (2021-07-13)

**Note:** Version bump only for package @blaze-react/dropdown







**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.17](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.16...v0.8.0-alpha.17) (2021-07-13)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.16](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.15...v0.8.0-alpha.16) (2021-07-13)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.15](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.14...v0.8.0-alpha.15) (2021-07-13)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.14](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.13...v0.8.0-alpha.14) (2021-07-13)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.13](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.12...v0.8.0-alpha.13) (2021-07-13)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.12](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.11...v0.8.0-alpha.12) (2021-07-13)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.11](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.10...v0.8.0-alpha.11) (2021-07-13)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.10](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.9...v0.8.0-alpha.10) (2021-07-13)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.9](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.8...v0.8.0-alpha.9) (2021-07-12)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.8](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.7...v0.8.0-alpha.8) (2021-07-12)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.7](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.6...v0.8.0-alpha.7) (2021-07-12)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.6](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.3...v0.8.0-alpha.6) (2021-07-09)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.3](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.2...v0.8.0-alpha.3) (2021-06-17)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.2](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.1...v0.8.0-alpha.2) (2021-06-17)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.1](https://github.com/thebyte9/blaze-components-react/compare/v0.8.0-alpha.0...v0.8.0-alpha.1) (2021-06-17)

**Note:** Version bump only for package @blaze-react/dropdown





# [0.8.0-alpha.0](https://github.com/thebyte9/blaze-components-react/compare/v0.6.2-alpha.0...v0.8.0-alpha.0) (2021-06-17)

**Note:** Version bump only for package @blaze-react/dropdown





## [0.6.2-alpha.0](https://github.com/thebyte9/blaze-components-react/compare/v0.6.1...v0.6.2-alpha.0) (2021-06-17)

**Note:** Version bump only for package @blaze-react/dropdown
