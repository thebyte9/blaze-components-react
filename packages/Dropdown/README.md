
## Description

Dropdown list to navigate between items.

## Usage

```js
<Dropdown label="Dropdown">
    <Link to="/">Settings</Link>
    <Link to="/">Sign out</Link>
    <Link to="/">Help</Link>
</Dropdown>
```

## API

##### Dropdown can receive a number of `props` as follow:

| NAME   | TYPE | DEFAULT | 
| :---  | :---:  | :---: | 
| label | String | Menu | 
| children | Single or array of nodes | No content |

