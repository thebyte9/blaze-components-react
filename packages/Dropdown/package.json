{
  "name": "@blaze-react/dropdown",
  "version": "0.6.0",
  "publishConfig": {
    "access": "public"
  },
  "description": "",
  "main": "./lib/index.js",
  "scripts": {
    "build": "../../node_modules/.bin/rimraf ./lib && ../../node_modules/.bin/tsc --jsx react --esModuleInterop ./src/* --outDir lib ",
    "build-storybook": "build-storybook -c .storybook -o docs-build",
    "lint": "../../node_modules/.bin/tslint -c ../../tslint.json src/**/*.tsx",
    "test": "jest",
    "test:coverage": "jest --coverage",
    "pre-publish": "npm run build && npm run lint && npm run test"
  },
  "keywords": [],
  "author": "",
  "license": "ISC",
  "devDependencies": {
    "@babel/cli": "^7.4.4",
    "@babel/core": "^7.4.5",
    "@blaze-react/blaze-components-theme": "^0.6.0"
  },
  "peerDependencies": {
    "prop-types": "^15.7.2",
    "react": "^16.9.0",
    "react-dom": "^16.9.0"
  },
  "babel": {
    "presets": [
      "@babel/preset-env",
      "@babel/preset-react"
    ]
  },
  "gitHead": "c7cc1d26b4046b800c727feb1dd283dbd833328c",
  "dependencies": {
    "@types/uuid": "^3.4.4"
  }
}
