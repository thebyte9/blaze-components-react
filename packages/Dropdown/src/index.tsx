import React, { useState, Fragment } from "react";
type DropdownProps = {
  label?: string,
  children?: any
};
const Dropdown: React.SFC<DropdownProps> = ({ label, children }) => {
  const [toggled, setToggled] = useState("close");
  const toggleMenu = () => {
    const menuStatus = toggled === "close" ? "open" : "close";
    setToggled(menuStatus);
  };
  return (
    <Fragment>
      <div className="more-menu__wrapper">
        <button
          onClick={toggleMenu}
          type="button"
          className="icon-button toggle"
        >
          {label}
          <i className="material-icons">more_vert</i>
        </button>
        <div className={`more-menu ${toggled}`}>
          <ul className="more-menu__list">
            <li className="more-menu__list-item">
              {React.Children.map(children, (child: any): JSX.Element => React.cloneElement(child))}
            </li>
          </ul>
        </div>
      </div>
    </Fragment>
  );
};
Dropdown.defaultProps = {
  label: "Menu",
  children: []
};
export default Dropdown;
