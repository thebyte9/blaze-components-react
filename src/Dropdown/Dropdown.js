
import React, { useState, Fragment } from 'react';
import PropTypes from 'prop-types';

export const DropdownItem = ({
  label,
  action
}) => (
  <li>
    <span onClick={action} role="button">{label}</span>
  </li>
);

DropdownItem.propTypes = {
  label: PropTypes.string,
  action: PropTypes.func
};
  
DropdownItem.defaultProps = {
  label: 'Empty label',
  action: () => {},
};

export const Dropdown = ({
  label,
  children
}) => {
  const [toggled, setToggled] = useState(false);

  return (
    <Fragment>
      <div onClick={() => setToggled(!toggled)} role="button">{label}</div>
      <ul>{toggled && children}</ul>
    </Fragment>
  );
};

Dropdown.propTypes = {
  label: PropTypes.string,
  children: PropTypes.oneOfType([PropTypes.arrayOf(PropTypes.node), PropTypes.node])
};

Dropdown.defaultProps = {
  label: 'No content',
  children: <li>No content</li>
};
