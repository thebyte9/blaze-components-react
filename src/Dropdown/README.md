
## Description

Dropdown list to navigate between items.

## Usage

```js
<Dropdown label="Dropdown">
    <Link href="/">Settings</Link>
    <Link href="/">Sign out</Link>
    <Link href="/">Help</Link>
</Dropdown>
```

## API

##### Dropdown can receive a number of `props` as follow:

| NAME   | TYPE | DEFAULT | Description |
| :---  | :---:  | :---: | ------- |
| label | String | Menu | Dropdown text |
| children | Single or array of nodes | No content | Any children components within the badge |

