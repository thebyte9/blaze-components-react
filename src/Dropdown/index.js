import { Dropdown, DropdownItem } from './Dropdown';

export default { Dropdown, DropdownItem };
