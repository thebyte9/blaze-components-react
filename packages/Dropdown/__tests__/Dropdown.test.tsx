import '@testing-library/jest-dom';

import { fireEvent, render, screen } from '@testing-library/react';

import Dropdown from '../src/Dropdown';
import React from 'react';

const component = (
  <Dropdown label="Dropdown">
    <a href="/">Settings</a>
    <a href="/">Sign out</a>
  </Dropdown>
);

describe('Dropdown component', () => {
  test('should be defined and renders correctly (snapshot)', () => {
    const { asFragment } = render(component);

    expect(asFragment).toMatchSnapshot();
  });
});
