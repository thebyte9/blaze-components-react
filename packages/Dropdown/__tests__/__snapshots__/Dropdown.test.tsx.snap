// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown component should be defined and renders correctly (snapshot) 1`] = `[Function]`;
